spec/common/modules/testapp_manager.js

```diff
@@ -1,15 +1,18 @@
 "use strict";
+/* global define, isNativescript, fetch */
 
 /* testapp module is responsible for setting up and tearing down apps in the test environment */
 define(['globals', 'browser-base64', 'ably'], function(ablyGlobals, base64, ably) {
 	var restHost = ablyGlobals.restHost || prefixDomainWithEnvironment('rest.ably.io', ablyGlobals.environment),
 			tlsPort  = ablyGlobals.tlsPort;
 
+
 	var isBrowser = (typeof(window) === 'object'),
 			httpReq   = httpReqFunction(),
-			toBase64  = base64Function(),
-			loadJsonData = isBrowser ? loadJsonDataBrowser : loadJsonDataNode,
-			testResourcesPath = (isBrowser && window.__karma__ && window.__karma__.start ? 'base/' : '') + 'spec/common/ably-common/test-resources/';
+			toBase64  =  base64Function(),
+			loadJsonData = isNativescript ? loadJsonNativescript : isBrowser ? loadJsonDataBrowser : loadJsonDataNode,
+			testResourcesPath = isNativescript ? '~/ably/ably-common/test-resources/' :
```

Comment:
- This should be using a relative path, not an absolute one -- we can't assume that the person running the tests happens to have a folder called `ably` in their home dir and has cloned the ably-common repo to it.

- the ably-js repo has ably-common as a submodule (it's in `spec/common/ably-common/test-resources/`), so as well as being a relative path, it should be a relative path to that instance of ably-common, not a separately cloned instance.

- This assumes the existance of a global variable called `isNativescript`. If we're not using nativescript, that variable isn't false, it just doesn't exist, so this will crash with a reference error.

- Minor point, but not a huge fan of nested ternaries